packages/colony-js-contract-client/src/classes/ContractClient.js

```
@@ -117,6 +119,35 @@ export default class ContractClient {
     return this.contract.callTransaction(functionName, args, options);
   }
 
+  /**
+   * Get logs from the contract with filter, return parsed event logs.
+   */
+  async getLogs(
+    filter?: LogFilter & { eventNames?: Array<string> } = {},
+  ): Promise<Array<Object>> {
+    const { eventNames = [], topics = [] } = filter;
+
+    // Get topics for the given eventNames
+    const extraTopics: Array<string> = eventNames
+      .filter(eventName => Object.keys(this.events).includes(eventName))
```

Maybe instead throw an error for events that don't exist? Otherwise it's not clear that I'm not getting what I asked for.